src/Protocol/Diaspora/Repository/DiasporaContact.php

```diff
@@ -0,0 +1,251 @@
+	 */
+	public function selectOne(array $condition, array $params = []): Entity\DiasporaContact
+	{
+		return parent::_selectOne($condition, $params);
+	}
+
+	/**
+	 * @param UriInterface $url
+	 * @return Entity\DiasporaContact
+	 * @throws HTTPException\NotFoundException
+	 */
+	public function selectOneByUrl(UriInterface $url): Entity\DiasporaContact
+	{
+		return $this->selectOne(['url' => (string)$url]);
+	}
+
+	public function existsByUri(string $uri): bool
+	{
+		return parent::exists(['`url` = ? OR `alias` = ? OR `addr` = ?', $uri, $uri, $uri]);
+	}
+
+	public function save(Entity\DiasporaContact $DiasporaContact): Entity\DiasporaContact
+	{
+		$fields = [
+			'url'               => (string)$DiasporaContact->url,
+			'uri-id'            => $DiasporaContact->uriId,
+			'guid'              => $DiasporaContact->guid,
+			'addr'              => $DiasporaContact->addr,
+			'alias'             => (string)$DiasporaContact->alias,
+			'nick'              => $DiasporaContact->nick,
+			'name'              => $DiasporaContact->name,
+			'given-name'        => $DiasporaContact->givenName,
+			'family-name'       => $DiasporaContact->familyName,
+			'photo'             => (string)$DiasporaContact->photo,
+			'photo-medium'      => (string)$DiasporaContact->photoMedium,
+			'photo-small'       => (string)$DiasporaContact->photoSmall,
+			'batch'             => (string)$DiasporaContact->batch,
+			'notify'            => (string)$DiasporaContact->notify,
+			'poll'              => (string)$DiasporaContact->poll,
+			'subscribe'         => (string)$DiasporaContact->subscribe,
+			'searchable'        => $DiasporaContact->searchable,
+			'pubKey'            => $DiasporaContact->pubKey,
+			'baseurl'           => (string)$DiasporaContact->baseurl,
+			'gsid'              => $DiasporaContact->gsid,
+			'updated'           => DateTimeFormat::utcNow(),
+			'interacting_count' => $DiasporaContact->interacting_count,
+			'interacted_count'  => $DiasporaContact->interacted_count,
+			'post_count'        => $DiasporaContact->post_count,
+		];
+
+		$this->db->insert(self::$table_name, $fields, Database::INSERT_UPDATE);
+
+		return $this->selectOneByUrl($DiasporaContact->url);
+	}
+
+	/**
+	 * Fetches a Diaspora profile from a given URI
+	 *
+	 * @param string  $uri    profile URI
+	 * @param boolean $update true = always update, false = never update, null = update when not found or outdated
+	 * @return Entity\DiasporaContact
+	 * @throws HTTPException\InternalServerErrorException
+	 * @throws HTTPException\NotFoundException
+	 * @throws \ImagickException
+	 */
+	public function getByUri(string $uri, ?bool $update = self::UPDATE_IF_MISSING_OR_OUTDATED): Entity\DiasporaContact
+	{
+		if (empty($uri)) {
+			throw new \InvalidArgumentException('Unexpected empty url parameter');
+		}
+
+		$dcontact = null;
+		if ($update !== self::ALWAYS_UPDATE) {
+			try {
+				$ref_update = new \DateTime('now - 1 month', new \DateTimeZone('UTC'));
```

Comment:
Please reintroduce the asynchronous update of a contact like it is done here: https://github.com/friendica/friendica/blob/develop/src/Model/FContact.php#L58-L64

Reply:
I set out to do that, but there's a logic problem with this snippet:
```
if (GServer::getNextUpdateDate(true, $person['created'], $person['updated'], false) < DateTimeFormat::utcNow()) {
```
The condition cannot be true since `GServer::getNextUpdateDate` always returns a date in the future.

Please advise @annando